src/ansys/speos/core/bsdf.py

```diff
@@ -774,7 +856,350 @@ def save(self, file_path: Union[Path, str], commit: bool = True) -> Path:
         else:
             file_name.file_name = str(file_path.parent / (file_path.name + ".anisotropicbsdf"))
         self._stub.Save(file_name)
-        return file_name.file_name
+        return Path(file_name.file_name)
+
+
+class SpectralBRDF(BaseBSDF):
+    """BSDF - Bidirectional scattering distribution function.
+
+    This class contains the methods and functions to load and edit existing Speos bsdf datasets.
+
+    Parameters
+    ----------
+    speos : ansys.speos.core.speos.Speos
+        Speos Object to connect to speos rpc server
+    file_path : Union[Path, str]
+        File path to bsdf file
+    """
+
+    def __init__(self, speos: Speos, file_path: Union[Path, str] = None):
+        super().__init__(
+            speos, spectral_bsdf__v1__pb2_grpc.SpectralBsdfServiceStub(speos.client.channel)
+        )
+        self._spectrum_incidence = [0, 0]
+        self._spectrum_anisotropy = [0, 0]
+        if file_path:
+            file_path = Path(file_path)
+            self._grpcbsdf = self._import_file(file_path)
+            self.brdf, self.btdf = self._extract_bsdf()
+            self._has_transmission = bool(self._btdf)
+            self._has_reflection = bool(self._brdf)
+            try:
+                self._stub.GetSpecularInterpolationEnhancementData(Empty())
+                self.__interpolation_settings = InterpolationEnhancement(
+                    bsdf=self,
+                    bsdf_namespace=spectral_bsdf__v1__pb2,
+                    index_1=None,
+                    index_2=None,
+                )
+            except grpc.RpcError:
+                self.__interpolation_settings = None
+        else:
+            self._transmission_spectrum, self._reflection_spectrum = None, None
+            self.__interpolation_settings = None
+
+    @property
+    def wavelength(self):
+        """List of all Wavelength in BRDF."""
+        r_wl = []
+        t_wl = []
+        if self.has_reflection:
+            for brdf in self.brdf:
+                r_wl.append(brdf.wavelength)
+            r_wl = list(set(r_wl))
+            r_wl.sort()
```

Comment:
list(set() already sort the value from small to large. no need to sort again

Reply:
solved that is remaining sorts from trying to fix an issue with unsorted data